internal/catalog/internal/types/validators.go

```diff
@@ -139,7 +139,9 @@ func validatePortName(name string) error {
 
 func validateProtocol(protocol pbcatalog.Protocol) error {
 	switch protocol {
-	case pbcatalog.Protocol_PROTOCOL_TCP,
+	case pbcatalog.Protocol_PROTOCOL_UNSPECIFIED,
+		// means pbcatalog.FailoverMode_FAILOVER_MODE_TCP
+	    pbcatalog.Protocol_PROTOCOL_TCP,
```

Comment:
~~ weird whitespace

Reply:
Yeah, I was following suit from [this](https://github.com/hashicorp/consul/blob/892ba52c56db7c3f1a17cbe5f158c1492c9673b3/internal/catalog/internal/types/failover_policy.go#L144) but agree it's a bit weird-looking